alibi_detect/utils/pytorch/__init__.py

```diff
@@ -16,5 +40,6 @@
     "predict_batch_transformer",
     "get_device",
     "quantile",
-    "zero_diag"
+    "zero_diag",
+    "TorchDataset"
```

Comment:
General point here: Where we have promoted (public!) imports up to a higher submodule (or just moved things around entirely), IMO we at a minimum need to document the changes properly in the release notes. Better yet we should raise `DeprecationWarning`'s if the old imports are used. 

An example is the old `from alibi_detect.utils.saving import save_detector` has been moved to `from alibi_detect.saving import save_detector`, however calling the former will still import `save_detector` and raise a `DeprecationWarning`.

Reply:
Me and @ascillitoe discussed options: 

The best way of providing depreciation warnings is to move the functionality in `alibi_detect.utils.pytorch.misc` to `alibi_detect.utils.pytorch._misc`. We then update the `__init__.py` file to import from `_misc.py`:

```py
get_device, quantile, zero_diag = import_optional(
    'alibi_detect.utils.pytorch._misc',
    names=['get_device', 'quantile', 'zero_diag']
)
```

In the `misc.py` file we replace the relevant functions with a depreciation warning similar to in the [config work](https://github.com/SeldonIO/alibi-detect/blob/release/0.10.0/alibi_detect/utils/saving.py). I think this makes sense but the result would be inconsistencies in what is private and what is public in a wider sense throughout detect. I could try and make this consistent but this would mean renaming lots of files with the underscore...

Reply:
IMO moving backend-specific files to private (to avoid imports without optional deps checks) might be a worthwhile exercise in a subsequent PR (to include in `0.10.0` release). However, personally I would only vote for this change if it doesn't screw up the git history for these files too much i.e. if we can get the git history to just track the filename change instead of registering it as a complete deletion and rewrite of each file.